### PR TITLE
Add website relation

### PR DESCRIPTION
**Pull request by Thanhphan1147**
Add a basic implementation of the website relation, simply sending over the unit's IP and port

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

**Review by javierdelapuente — DISMISSED** [review]
Could a little scenario test or similar be useful here?

**Comment by Thanhphan1147** [comment]
> Could a little scenario test or similar be useful here?

Indeed, I'll add it in a follow-up PR

File: src/grafana_dashboards/haproxy.json [hunk 1]
```diff
@@ -12099,4 +12099,4 @@
   "uid": "rEqu1u5ue",
```
Comment (amandahla) [inline src/grafana_dashboards/haproxy.json:12099]: out of scope: uid should be null or absent

**Review by amandahla — DISMISSED** [review]

File: src/grafana_dashboards/haproxy.json (hunk 1 above)
Comment (Thanhphan1147) [reply src/grafana_dashboards/haproxy.json:12099]: Got it, will update it in a follow-up PR

File: src/http_interface.py [hunk 2]
```diff
@@ -174,6 +174,28 @@ def get_services_definition(self) -> dict:
...
+
+        Args:
+            event: relation-changed event.
+        """
+        event.relation.data[self.charm.unit].update({"hostname": self.bind_address, "port": 80})
+
+    # We add a placeholder implementation of this method because of parent class
+    def _on_relation_changed(self, _: RelationChangedEvent) -> None:
+        """Handle relation-changed event."""
+        logger.warning("Nothing to do for relation-changed hook of website relation, skipping.")
```
Comment (gregory-schiano) [inline src/http_interface.py]: Not sure a warning is required here, that's nothing abnormal, "info" or "debug" are better suited.
Same for relation broken

File: src/http_interface.py [hunk 3]
```diff
@@ -174,6 +174,28 @@ def get_services_definition(self) -> dict:
...
+
+    # We add a placeholder implementation of this method because of parent class
+    def _on_relation_broken(self, _: RelationBrokenEvent) -> None:
```
Comment (gregory-schiano) [inline src/http_interface.py]: Shouldn't the databag be emptied in this case ?

File: src/charm.py [hunk 4]
```diff
@@ -72,7 +78,8 @@ def __init__(self, *args: typing.Any):
         self.certificates = TLSCertificatesRequiresV3(self, TLS_CERT_RELATION)
         self._tls = TLSRelationService(self.model, self.certificates)
         self._ingress_provider = IngressPerAppProvider(charm=self, relation_name=INGRESS_RELATION)
-        self.http_provider = HTTPProvider(self, REVERSE_PROXY_RELATION)
+        self.http_requirer = HTTPRequirer(self, REVERSE_PROXY_RELATION)
+        self.http_provider = HTTPProvider(self, WEBSITE_RELATION)
```
Comment (gregory-schiano) [inline src/charm.py]: As discussed on MM I'd rather call these variables  "website_provider" and "reverseproxy_requirer"

File: src/http_interface.py (hunk 3 above)
Comment (Thanhphan1147) [reply src/http_interface.py]: Makes sense, added the code to clear the databag

File: src/charm.py (hunk 4 above)
Comment (Thanhphan1147) [reply src/charm.py]: Got it, I changed the name of these attributes to  `website_provider` and `reverseproxy_requirer`

**Review by yanksyoon — APPROVED** [review]

File: src/http_interface.py (hunk 2 above)
Comment (Thanhphan1147) [reply src/http_interface.py]: updated to "debug"

**Review by javierdelapuente — APPROVED** [review]

**Review by gregory-schiano — APPROVED** [review]

**Pull request merged**